src/System.Private.CoreLib/src/System/Threading/ClrThreadPool.CpuUtilizationReader.Windows.cs

```diff
@@ -0,0 +1,77 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+
+namespace System.Threading
+{
+    internal partial class ClrThreadPool
+    {
+        private class CpuUtilizationReader
+        {
+            private struct ProcessCpuInformation
+            {
+                public long idleTime;
+                public long kernelTime;
+                public long userTime;
+                public int numberOfProcessors;
+                public Interop.mincore.SystemProcessorPerformanceInformation[] usageBuffer;
+            }
+
+            private ProcessCpuInformation _cpuInfo = new ProcessCpuInformation();
+
+            public CpuUtilizationReader()
+            {
+                _cpuInfo.numberOfProcessors = ThreadPoolGlobals.processorCount;
+
+                _cpuInfo.usageBuffer = new Interop.mincore.SystemProcessorPerformanceInformation[ThreadPoolGlobals.processorCount];
+                GetCpuUtilization(); // Call once to initialize the usage buffer
+            }
+
+            private unsafe int GetCpuUtilization()
+            {
+                fixed (Interop.mincore.SystemProcessorPerformanceInformation* buffer = _cpuInfo.usageBuffer)
+                {
+                    int status = Interop.mincore.QuerySystemInformation(Interop.mincore.SystemInformationClass.SystemProcessorPerformanceInformation,
```

Comment:
Looking at it more, can we use the GetSystemTimes function instead? It gives us the three values we're interested in summed across all processors. Since we're not checking affinity that might be better.